src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java

```diff
@@ -28,6 +33,11 @@ public static void setupResearches() {
         if (alreadyRan) {
             throw new UnsupportedOperationException("Researches can only be registered once!");
         }
+        
+        plugin = SlimefunPlugin.instance();
+        if (plugin == null) {
+            throw new UnsupportedOperationException("Plugin cannot be null!");
+        }
```

Comment:
Still, unnecessary. If you really wanna validate the input, add the plugin as a method parameter and do `Validate.notNull()` but not this.

I think using `SlimefunPlugin.instance()` like before was just fine too.

Reply:
I just thought that getting using a field would be more light-weight than running the method over 250 times.